src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -82,10 +93,8 @@ public TrieNode FromCachedRlpOrUnknown(Keccak hash)
                     }
 
                     // we returning a copy to avoid multithreaded access
-                    trieNode = new TrieNode(NodeType.Unknown, hash, trieNode.FullRlp);
+                    trieNode = new TrieNode(trieNode.NodeType, hash, trieNode.FullRlp);
                     trieNode.ResolveNode(_trieStore);
-                    trieNode.Keccak = hash;
```

Comment:
Is this change required for the background prune? If not I recommend moving to a different PR.  Seems dangerous.

Reply:
@LukaszRozmej Is this answered/addressed?

Reply:
I am trying to figure out why https://github.com/NethermindEth/nethermind/blob/f14da669b1a70c13dabaa8dd81f9a3d56ed98644/src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs#L493-L496 is happening and this in one of my leads, that we overwrite the actual proper hash here as ResolveNode returned something different.